container_linux.go

```diff
@@ -169,21 +168,58 @@ func (c *linuxContainer) newInitProcess(p *Process, cmd *exec.Cmd, parentPipe, c
 			cmd.SysProcAttr.Credential = &syscall.Credential{}
 		}
 	}
-	cmd.Env = append(cmd.Env, t)
 	cmd.SysProcAttr.Cloneflags = cloneFlags
+
+	// set init process environment
+	env := []string{"_LIBCONTAINER_INITTYPE=standard"}
+	var joinNamespaces configs.Namespaces
+	var nsPaths []string
+	var doClone bool
+	for _, ns := range c.config.Namespaces {
+		if ns.Path != "" {
+			joinNamespaces = append(joinNamespaces, ns)
+			nsPaths = append(nsPaths, ns.Path)
+			if ns.Type == configs.NEWPID {
+				doClone = true
+			}
```

Comment:
You can't to enter into namespaces in a random order. For example, when you enter into a userns, you may lose permissions to enter in other namespaces.